invenio_rdm_records/collections/service.py

```diff
@@ -131,3 +162,43 @@ def read_logo(self, identity, slug):
         if _exists:
             return url_for("static", filename=logo_path)
         raise LogoNotFoundError()
+
+    def read_many(self, identity, ids_=None, depth=2):
+        """Get many collections."""
+        self.require_permission(identity, "read")
+
+        ids_ = ids_ or []
+        res = self.collection_cls.resolve_many(ids_, depth=depth)
+        return CollectionList(
+            identity, res, self.collection_schema, None, self.links_item_tpl
+        )
+
+    def read_all(self, identity, depth=2):
+        """Get all collections."""
+        self.require_permission(identity, "read")
+        res = self.collection_cls.resolve_all(depth=depth)
+        return CollectionList(
+            identity, res, self.collection_schema, None, self.links_item_tpl
+        )
+
+    def search_records(self, identity, collection_or_id, params=None):
+        """Search records in a collection."""
+        params = params or {}
+
+        if isinstance(collection_or_id, int):
+            collection = self.collection_cls.resolve(id_=collection_or_id)
+        else:
+            collection = collection_or_id
+
+        params.update({"collection_id": collection.id})
```

Comment:
I am puzzled by fetching the collection query inside the community service [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/community_records/service.py#L71).
It feels like a specific case, and if in the future when need to add another `extra_filter` there, we will copy the pattern and add an extra if/else there.
Why not fetching here the collection query, and injecting it in the `search` as an extra filter?
It also feels like we are fetching multiple times the collection object around.

Reply:
I think this is a really good point and I had the same feeling that it could go just in one place. Plus, collections are not bound to communities at its core so the search should be possible without going through the community records service.

For this change, however, we need a new endpoint inside collections to expose the collection service search method

Reply:
We discussed IRL and we were on the same page: the collection's records search is better fitted in the collection service. Therefore, I added a new commit with the required changes. 

In order not to block the development, both commits of this PR are working so we can split them if needed.